pkg/providers/v1/awsnode/identifier.go

```diff
@@ -0,0 +1,11 @@
+package awsnode
+
+import "github.com/aws/aws-sdk-go/aws"
+
+// NodeID is the ID of a node used to uniquely identify that node within an AWS service
+type NodeID string
```

Comment:
I think this could be more useful as an interface, we have too many string alias types in here already 😛 

Something like:
```
type AWSProviderID interface {
    ComputeID() string
}
```

Then we can implement an `EC2ProviderID`, `FargateProviderID`, etc.

The func-s in the `Variant` interface can accept an `AWSProviderID` and cast to the relevant type as necessary, `providerID.(EC2ProviderID)`

Reply:
hmm I actually am not quite following the need for an interface here. While the NodeID might have different formats, I dont see why there would be different implementations required for those different formats. Along that same line, while it sounds potentially useful to have strongly typed NodeID's, there is currently not a need for it — the casting you mention wouldnt actually be necessary anywhere. That to me signals that an interface may be overkill, but perhaps Im not thinking about a future use-case where having an interface would be ideal. 

So as it stands now, I actually think the ergonomics of the string alias type is fits the needs quite well. Let me know your thoughts.

Reply:
also, much less of a concern, but moving to an interface would substantially increase the size of the changes in this commit.